xarray/core/datatree_io.py

```diff
@@ -25,17 +39,23 @@ def _create_empty_netcdf_group(filename, group, mode, engine):
 
 def _datatree_to_netcdf(
     dt: DataTree,
-    filepath,
-    mode: str = "w",
-    encoding=None,
-    unlimited_dims=None,
+    filepath: str | PathLike,
+    mode: NetcdfWriteModes = "w",
+    encoding: Mapping[str, Any] | None = None,
+    unlimited_dims: Mapping | None = None,
     **kwargs,
 ):
+    """This function creates an appropriate datastore for writing a datatree to
+    disk as a netCDF file.
+
+    See `DataTree.to_netcdf` for full API docs.
+    """
+
     if kwargs.get("format", None) not in [None, "NETCDF4"]:
         raise ValueError("to_netcdf only supports the NETCDF4 format")
 
     engine = kwargs.get("engine", None)
-    if engine not in [None, "netcdf4", "h5netcdf"]:
+    if engine not in [None, *get_args(T_DataTreeNetcdfEngine)]:
```

Comment:
No stopper but I'm not sure I like the `kwargs.get`-style here, `engine` wont be typed and might hide typing issues downstream. Is `engine` correctly narrowed from `Any` after this if-case? 

If we want good typing coverage it's better to prefer keyword-only arguments rather than using `kwargs.get`.

Reply:
Interesting - yeah. I took a quick look at `to_netcdf` in [xarray/backends/api.py](https://github.com/pydata/xarray/blob/main/xarray/backends/api.py#L1241), and it looks like making these explicitly named with defaults would be consistent with that, too.

Looks like the potential list of arguments is: `format`, `engine`, `group` and `compute` (with `group` and `compute` for `_datatree_to_zarr`). These changes would also presumably require similar updates to [DataTree.to_netcdf](https://github.com/pydata/xarray/blob/main/xarray/core/datatree.py#L1477) and `DataTree.to_zarr(https://github.com/pydata/xarray/blob/main/xarray/core/datatree.py#L1516)`.

Just to check @TomNicholas - am I missing a reason for keeping the `kwargs.get` implementation?

Reply:
I can't see a valid reason not to replace them, but I'm not 100% familiar.

Reply:
I like the idea of making them consistent with the other `to_netcdf` functions in `xarray/backends.api.py`. I'll make this change and push it up as I get chance.

Reply:
I've made these changes in [this commit](https://github.com/pydata/xarray/pull/9011/commits/e968631628b499ef056a111af5e5715fa13178b7)